docs/data/date-pickers/date-picker/examples/DateCalendar/DesktopDatePicker/DateCalendarStyledComponents.styling.tsx

```diff
@@ -0,0 +1,23 @@
+import * as React from 'react';
+import { DesktopDatePicker } from '@mui/x-date-pickers/DesktopDatePicker';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { styled } from '@mui/material/styles';
+
+const StyledDesktopDatePicker = styled(DesktopDatePicker)(() => ({
+  '& .MuiDateCalendar-root': {
+    height: 'fit-content',
+    maxHeight: '400px',
+    '& *': { fontFamily: 'Arial' },
+  },
+}));
+
+/* You will need to use the `disablePortal` prop from the popper 
```

Comment:
Sorry if I have confused you, but I'm not sure that we should be endorsing this usage as a general rule.
I think that we either have to choose a component that can be easily overridden both via `themeOverrides` as well as `slots` (i.e. `CalendarHeader`) or use different example in case of `Desktop` or `Mobile` picker. 🤔

Reply:
Thanks! I added examples for styling with `sx`, because it is probably better in the case of the `DesktopDatePicker`. I didn't remove the styled component examples just yet, because I'd like to get your input on a 'recommendation system' I'm experimenting with (I'm calling it that because I have no idea what else to call it) 😅 
We can mark the approaches that we consider the best or the ones we don't like this: 
<img width="449" alt="image" src="https://github.com/mui/mui-x/assets/72460825/1906868d-803e-4932-8a42-a02747bfe656">
What do you think about this? 

cc. @flaviendelangle

Reply:
That's a good idea.
But wouldn't it end up being basically the same story for every component, that we recommend the `Custom theme` option? 🤔